src/screen.rs

```diff
@@ -1078,10 +1078,30 @@ impl Screen {
         self.need_clear_lines = false;
         self.need_clear_screen = false;
 
+        // Compare the number of lines in the (new) left prompt compared to the
+        // current prompt before it's dropped so that the lines can be cleared
+        // appropriately later (see #11875). Avoid unnecessary clearing by only
```

Comment:
LGTM, nice work. Add a changelog entry for 4.2 (i.e. "fish ?.?.?") if you want.
(Not sure if it's a good idea to add this to a patch release)

Reply:
@krobelus Sure, I just force-pushed this and the other recommended changes. I put it under "Notable improvements and fixes" as that seemed like the most appropriate section from previous major releases, but TBH it felt odd calling this "notable." FWIW this feels like a minor regression fix to me, more appropriate to a patch release, but I'd wager I'm not particularly well-calibrated on that front...

Reply:
Hmm, #11911 might be the "right" fix for this bug, but seems bigger. (NB: I didn't look at it super closely) I guess if that one is the way forward but makes sense for a minor version release, maybe this one can work as a hold-over in a patch release until then.

In case it's helpful (for cherry picking, etc.), I put the changes to the tests in a dedicated commit.